step-plans/step-plans-simple-yaml-parser/src/test/resources/step/plans/simple/test-simple.plan.yml

```diff
@@ -0,0 +1,27 @@
+name: "test plan"
+root:
+  Sequence:
+    name: Sequence
+    children:
+      - If:
+          nodeName: IfBlock
+          condition:
+            expression: "controllerSettings.getSettingByKey('housekeeping_enabled').getValue()=='true'"
```

Comment:
@iegorov777 For some special cases it would be useful to have a simplified syntax that isn't directly derived from the model of the artefact. Does the implementation of the parser allow this? in this case we could for instance have the need to simplify as follow:
- If:
    condition: "controllerSettings.getSettingByKey('housekeeping_enabled').getValue()=='true'"
    then:
      - Assert...
      - ...

or even

- If: "controllerSettings.getSettingByKey('housekeeping_enabled').getValue()=='true'"
  then: 
    - Assert...

Would this be possible in theory? (no need to change it now)

Reply:
@jeromecomte 
The main limitation, which restrics this kind of changes is the Json schema used for user-friendly input in text editor in IDE.
Because of this limitation, we need to have some regular structure like

```
root:
   {root_artefact_name}:
        children:
            - {artefact_name}:
               children:
                  - {artefact_name_1_1}
                  - ...
            - {artefact_name_2}:
            - {artefact_name_3}:
            - ....
```

Otherwise, if we don't have the fixed elements (nodes) containing the node name (`root_artefact_name` and `artefact_name`), on-the-fly suggestions for artefact names in IDEA don't work (the editor can't resolve a fixed set of suggested options from Json Schema).